src/plugins/strapi-plugin-internal-links/admin/src/components/form/source-select/index.tsx

```diff
@@ -0,0 +1,94 @@
+import { useIntl } from 'react-intl';
+import debounce from 'lodash/debounce';
+import { useFetchClient } from '@strapi/helper-plugin';
+import { Flex, FieldLabel } from '@strapi/design-system';
+
+import { Combobox, IReactSelectValue } from '../../Combobox';
+import getTrad from '../../../utils/get-trad';
+import { Label } from '../../label';
+
+import { fetchSource } from '../../../api/fetch-source';
+const SEARCH_DEBOUNCE_MS = 150;
+
+interface Props {
+	selectedValue?: string;
+	externalApiUrl?: string;
+	externalApiValueMapping?: string;
+	externalApiLabelMapping?: string;
+	onChange: (item?: any) => void;
+}
+
+export const ExternalApiSearch = ({
+	onChange,
+	selectedValue,
+	externalApiUrl,
+	externalApiLabelMapping,
+	externalApiValueMapping
+}: Props) => {
+	const { formatMessage } = useIntl();
+	const fetchClient = useFetchClient();
+
+	const mappedSelectedValue = mapSelectItem(selectedValue);
+
+	const getItems = async (inputValue?: string): Promise<IReactSelectValue[]> => {
+		const externalItems = await fetchSource({ fetchClient }, externalApiUrl, inputValue);
+		if (!externalItems || !externalApiLabelMapping || !externalApiValueMapping) {
+			return [];
+		}
+
+		const mappedData = externalItems.data.map((item: any) => ({
+			value: item[externalApiValueMapping],
+			label: item[externalApiLabelMapping]
+		}));
+
+		return mappedData;
+	};
+
+	const handleChange = (item?: IReactSelectValue) => onChange(item);
+
+	const debouncedFetch = debounce((searchTerm, callback) => {
```

Comment:
Ik zou dit liever een duidelijke naam willen hebben, je hoeft in de naam niet te zetten dat iets een debounce is. Meer iets wat lijkt op getAllOptions ofzo